lib/ansible/modules/cloud/oneandone/oneandone_server.py

```diff
@@ -0,0 +1,644 @@
+        if wait:
+            operation_completed = False
+            wait_timeout = time.time() + wait_timeout
+            while wait_timeout > time.time():
+                time.sleep(5)
+                server = oneandone_conn.get_server(server['id'])  # refresh
+                server_state = server['status']['state']
+                if state == 'stopped' and server_state == 'POWERED_OFF':
+                    operation_completed = True
+                    break
+                if state == 'running' and server_state == 'POWERED_ON':
+                    operation_completed = True
+                    break
+            if not operation_completed:
+                module.fail_json(
+                    msg="Timeout waiting for server %s to get to state %s" % (
+                        server_id, state))
+
+        changed = True
+        server = _insert_network_data(server)
+
+    return (changed, server)
+
+
+def _auto_increment_hostname(count, hostname):
+    """
+    Allow a custom incremental count in the hostname when defined with the
+    string formatting (%) operator. Otherwise, increment using name-01,
+    name-02, name-03, and so forth.
+    """
+    if '%' not in hostname:
+        hostname = "%s-%%01d" % hostname
+
+    return [
+        hostname % i
+        for i in xrange(1, count + 1)
+    ]
+
+
+def _auto_increment_description(count, description):
+    """
+    Allow the incremental count in the description when defined with the
+    string formatting (%) operator. Otherwise, repeat the same description.
+    """
+    if '%' in description:
+        return [
+            description % i
+            for i in xrange(1, count + 1)
+        ]
+    else:
+        return [description] * count
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            auth_token=dict(
+                type='str',
+                default=os.environ.get('ONEANDONE_AUTH_TOKEN'),
+                no_log=True),
+            hostname=dict(type='str'),
+            description=dict(type='str'),
+            appliance=dict(type='str'),
+            fixed_instance_size=dict(type='str'),
+            vcore=dict(type='int'),
+            cores_per_processor=dict(type='int'),
+            ram=dict(type='float'),
+            hdds=dict(type='list'),
+            count=dict(type='int', default=1),
+            ssh_key=dict(type='raw', default=None),
+            auto_increment=dict(type='bool', default=True),
+            server=dict(type='str'),
+            datacenter=dict(
+                choices=DATACENTERS,
+                default='US'),
+            private_network=dict(type='str'),
+            firewall_policy=dict(type='str'),
+            load_balancer=dict(type='str'),
+            monitoring_policy=dict(type='str'),
+            wait=dict(type='bool', default=True),
+            wait_timeout=dict(type='int', default=600),
+            state=dict(type='str', default='present'),
```

Comment:
please add a list of choices here: 

~~~diff
-            state=dict(type='str', default='present'),
+            state=dict(type='str', default='present', choices=['present', 'absent', 'running', 'stopped']),
~~~